src/arch/riscv/traps.S

```diff
@@ -98,50 +98,43 @@ trap_entry:
   csrr x1,  sepc
   STORE   x1, (33*REGBYTES)(t0)
 
-#ifdef CONFIG_FASTPATH
-  li t3, SYSCALL_REPLY_RECV
-  beq a7, t3, fp_reply_recv
-
-  li t3, SYSCALL_CALL
-  beq a7, t3, fp_call
-#endif
-
   /* Check if it's an interrupt */
   bltz s0, interrupt
 
-  li   s4, 8       /* ratified priv has value 8 for ecall exception */
+  /* ratified priv has value 8 for ecall exception */
+  li   s4, 8
   bne  s0, s4, exception
 
-syscall:
+  /* If it's not an exception or an interrupt it's a syscall */
```

Comment:
That comment is a bit misleading actually, better simply say "handle syscall" now.